weather_dl/download_pipeline/util.py

```diff
@@ -42,3 +44,17 @@ def retry_with_exponential_backoff(fun):
         retry_filter=_retry_if_valid_input_but_server_or_socket_error_and_timeout_filter,
         clock=clock,
     )(fun)
+
+
+# TODO(#245): Group with common utilities (duplicated)
+def ichunked(iterable: t.Iterable, n: int) -> t.Iterator[t.Iterable]:
```

Comment:
Where is this from? If you implemented it yourself, please add a basic test. If you copied or adapted it from another library, it needs to go in a `third_party` sub-directory and preserve the original license.

Reply:
This came from here: https://github.com/google/weather-tools/blob/06cbc22a931c44ef9e82f18082ed29a7fca1e31c/weather_mv/loader_pipeline/util.py#L153

It was developed in house. Tests live next to where the routine was originally developed. My reasoning for not copying the tests here also war primarily that I wanted to minimize duplicate code.

I wanted to remove the duplication and instead extract out a common module (#245), but that would complicate this CL. 

How about this for a resolution: I copy with tests over with the same TODO, and clean it up when we get to the issue?

Reply:
Ah, gotcha. I missed this because I didn't read the linked GitHub issue!

You could also just add a brief comment like "duplicate from file XYZ, which has tests", but this is fine, too!